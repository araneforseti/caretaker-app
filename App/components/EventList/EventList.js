import React, {Component} from 'react';
import {
    FlatList,
    StyleSheet,
    Text,
    View
} from 'react-native';

export default class EventList extends Component {
    render() {
        return (
            <View style={styles.container}>
            <Text style={styles.title}>{this.props.title}</Text>
            <FlatList
               data={this.props.events}
               renderItem={({item}) => <Text style={styles.item}>{item.date}: {item.name} - {item.volunteer}</Text>}
            />
            </View>
        );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 24,
    textAlign: 'center',
    margin: 15,
  },
  item: {
    fontSize: 20,
    textAlign: 'left',
    margin: 13,
  },
});
