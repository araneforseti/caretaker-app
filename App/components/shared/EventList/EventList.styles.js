import { StyleSheet } from 'react-native';
import theme from '../../../config/theme';

export default styles = StyleSheet.create({
  container: {
    flex: 1
  },
  list: {
    backgroundColor: '#FFF'
  }
});
