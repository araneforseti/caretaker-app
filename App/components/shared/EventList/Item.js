import React from 'react';
import {
  View,
  TouchableOpacity
} from 'react-native';
import DateDisplay from '../DateDisplay';
import EventSummaryBox from '../EventSummaryBox';

import styles from './Item.styles';

export default class Item extends React.PureComponent {
  _onPress = () => this.props.onPressItem(this.props.id);

  render() {
    const {
      startDate,
      name,
      volunteer
    } = this.props;

    return <TouchableOpacity onPress={this._onPress}>
      <View style={styles.container}>
        <DateDisplay date={startDate} />
        <EventSummaryBox name={name} volunteer={volunteer} />
      </View>
    </TouchableOpacity>;
  }
}
