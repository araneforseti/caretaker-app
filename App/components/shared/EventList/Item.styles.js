import { StyleSheet } from 'react-native';
import theme from '../../../config/theme';

export default styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    marginTop: 8,
    marginBottom: 0,
    marginLeft: 8,
    marginRight: 8
  }
});
