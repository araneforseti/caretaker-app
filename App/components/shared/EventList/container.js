import React from 'react';

export default container = (EventList) => class extends React.Component {
  constructor() {
    super();
    this.state = {
      modalVisible: false,
      selectedEvent: null
    };
  }

  _onPressItem = (eventId) => {
    let newState = this.state;
    newState.modalVisible = true;
    newState.selectedEvent = this.props.events.find((event) => event.id === eventId)
    this.setState(newState);
  };

  _onCloseModal = () => {
    let newState = this.state;
    newState.modalVisible = false;
    this.setState(newState);
  }

  render() {
    const {
      modalVisible,
      selectedEvent
    } = this.state;

    const {
      events
    } = this.props;

    return <EventList
      modalVisible={modalVisible}
      selectedEvent={selectedEvent}
      onPressItem={this._onPressItem}
      onCloseModal={this._onCloseModal}
      events={events}
    />;
  }
}
