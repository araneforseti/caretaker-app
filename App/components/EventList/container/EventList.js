import React, {Component} from 'react';
import EventGateway from '../gateway.js';
import EventList from '../EventList.js';


export default class EventListContainer extends Component {
  constructor() {
    super();
    this.gateway = new EventGateway();
    this.state = {
      title: 'Upcoming Events',
      events: []
    };
  }

  componentDidMount() {
    this.setState({events: this.gateway.getAll()});
  }

  render() {
    return <EventList title={this.state.title} events={this.state.events} />;
  }
}
