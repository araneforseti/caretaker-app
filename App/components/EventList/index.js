import React, {Component, PureComponent} from 'react';
import {
    FlatList,
    StyleSheet,
    Text,
    TouchableOpacity,
    View
} from 'react-native';
import {ContainerFor} from './container.js';
import EventModal from '../shared/EventModal/';

class MyListItem extends PureComponent {
    _onPress = () => {
        this.props.onPressItem(this.props.item);
    };

    render() {
        item = this.props.item;
        return (
            <TouchableOpacity onPress={this._onPress}>
                <View style={styles.button}>
                   <Text style={styles.item}>{item.date}: {item.name} - {item.volunteer}</Text>
                </View>
            </TouchableOpacity>
        );
    }
}

class EventList extends Component {

    _renderItem = ({item}) => (
        <MyListItem
            id={item.key}
            item={item}
            onPressItem={this.props.onPressItem}
            title={item.title}
        />
    );

    render() {
        return (
            <View style={styles.container}>
                <EventModal
                    event={this.props.selectedEvent}
                    visible={this.props.visibleModal}
                    close={this.props.onCloseModal}
                />
                <Text style={styles.title}>Upcoming Events</Text>
                <FlatList
                   data={this.props.events}
                   renderItem={this._renderItem}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 24,
    textAlign: 'center',
    margin: 15,
  },
  item: {
    fontSize: 20,
    textAlign: 'left',
    margin: 13,
  },
  modalContent: {
    margin: 13,
  },
  modalTitle: {
    fontSize: 24,
    textAlign: 'center',
    margin: 15
  },
  modalInfo: {
    fontSize: 20,
    textAlign: 'left',
    margin: 10
  }
});

export default ContainerFor(EventList);
