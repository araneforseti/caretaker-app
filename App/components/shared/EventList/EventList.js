import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  FlatList
} from 'react-native';
import EventModal from '../EventModal';

import Item from './Item';
import styles from './EventList.styles';
import container from './container';

const renderItem = (onPressItem, {item}) => <Item
  id={item.id}
  onPressItem={onPressItem}
  title={item.title}
  startDate={new Date(item.startDate)}
  endDate={new Date(item.startDate)}
  date={item.date}
  name={item.name}
  volunteer={item.volunteer}
/>;

function EventList( {
  selectedEvent,
  modalVisible,
  onCloseModal,
  onPressItem,
  events
} ) {
  return <View style={styles.container}>
    <EventModal
      event={selectedEvent}
      visible={modalVisible}
      onClose={onCloseModal}
    />
    <FlatList
       style={styles.list}
       data={events}
       dataWhichCouldChange={events.map((event) => event.volunteer)}
       renderItem={item => renderItem(onPressItem, item)}
       keyExtractor={(item) => item.id}
       removeClippedSubviews={true}
    />
  </View>;
}

EventList.propTypes = {
  events: PropTypes.array,

//From container
  selectedEvent: PropTypes.object,
  modalVisible: PropTypes.bool,
  onCloseModal: PropTypes.func,
  onPressItem: PropTypes.func
};

export default container(EventList);
