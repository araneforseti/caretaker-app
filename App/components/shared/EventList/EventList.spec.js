import React from 'react';
import renderer from 'react-test-renderer';
import EventList from './EventList';

describe('<EventList />', () => {
  const Jul_10_2017 = 1499662800000;
  const Jul_11_2017 = 1499749200000;

  let props;
  const eventList = () => renderer.create(<EventList {...props} />);

  beforeEach(() => {
    const eventA = {
      id: 1,
      name: 'Chemotherapy',
      startDate: Jul_10_2017,
      endDate: Jul_11_2017,
      date: 'Jul 10',
      volunteer: 'The Dude'
    };
    const eventB = {
      id: 2,
      name: 'Toxicity Check',
      startDate: Jul_10_2017,
      endDate: Jul_11_2017,
      date: 'Jul 10',
      volunteer: 'Waluigi'
    };

    props = {
      events: [eventA, eventB],
      selectedEvent: eventB,
      modalVisible: true,
      onCloseModal: jest.fn(),
      onPressItem: jest.fn()
    };
  });

  it('should render correctly', () => {
    expect(eventList()).toMatchSnapshot();
  });

});
