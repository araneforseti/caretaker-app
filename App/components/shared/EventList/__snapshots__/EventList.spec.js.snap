// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventList /> should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "date": "Jul 10",
          "endDate": 1499749200000,
          "id": 1,
          "name": "Chemotherapy",
          "startDate": 1499662800000,
          "volunteer": "The Dude",
        },
        Object {
          "date": "Jul 10",
          "endDate": 1499749200000,
          "id": 2,
          "name": "Toxicity Check",
          "startDate": 1499662800000,
          "volunteer": "Waluigi",
        },
      ]
    }
    dataWhichCouldChange={
      Array [
        "The Dude",
        "Waluigi",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onViewableItemsChanged={undefined}
    removeClippedSubviews={true}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 0,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                10
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                Mon
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#CC5D5D",
                    "borderRadius": 2,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "flex": 1,
                    "padding": 4,
                  },
                  false,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#CC5D5D",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    false,
                  ]
                }
              >
                Chemotherapy
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#CC5D5D",
                    },
                    Object {},
                    false,
                    false,
                  ]
                }
              >
                The Dude
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 0,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                10
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                Mon
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#CC5D5D",
                    "borderRadius": 2,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "flex": 1,
                    "padding": 4,
                  },
                  false,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#CC5D5D",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    false,
                  ]
                }
              >
                Toxicity Check
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#CC5D5D",
                    },
                    Object {},
                    false,
                    false,
                  ]
                }
              >
                Waluigi
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
